pkg/services/oracle/nodes.go

```diff
@@ -13,17 +15,8 @@ func (o *Oracle) UpdateOracleNodes(oracleNodes keys.PublicKeys) {
 	o.accMtx.Lock()
 	defer o.accMtx.Unlock()
 
-	old := o.oracleNodes
-	if isEqual := len(old) == len(oracleNodes); isEqual {
-		for i := range old {
-			if !old[i].Equal(oracleNodes[i]) {
-				isEqual = false
-				break
-			}
-		}
-		if isEqual {
-			return
-		}
+	if slices.EqualFunc(o.oracleNodes, oracleNodes, (*keys.PublicKey).Equal) {
+		return
```

Comment:
The same should be implemented for Notary, I'm not sure why we don't have this check for Notary service.

Reply:
Because of m/n vs 1/n? There is a check for the key presence.